test/ns_test.go

```diff
@@ -0,0 +1,132 @@
+package telepilot_test
+
+import (
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	pb "go.creack.net/telepilot/api/v1"
+)
+
+func TestPIDNamespace(t *testing.T) {
+	t.Parallel()
+
+	ts, ctx := newTestServer(t)
+
+	t.Run("ps", func(t *testing.T) {
+		t.Parallel()
+
+		jobID, err := ts.alice.StartJob(ctx, "ps", []string{"-e"})
+		noError(t, err, "Start job.")
+		t.Cleanup(func() { noError(t, ts.alice.StopJob(ctx, jobID), "Cleanup stop job.") })
+
+		w := &strings.Builder{}
+		noError(t, ts.alice.StreamLogs(ctx, jobID, w), "Stream logs.")
+		// We expect only 2 lines: the headers and the lone process in the namesapce.
+		if l := len(strings.Split(strings.TrimSpace(w.String()), "\n")); l != 2 {
+			t.Fatalf("Unexpected line count for `ps -e`: %d.\n%s\n", l, w)
+		}
+	})
+	t.Run("getpid", func(t *testing.T) {
+		t.Parallel()
+
+		jobID, err := ts.alice.StartJob(ctx, "sh", []string{"-c", "echo $$"})
+		noError(t, err, "Start job.")
+		t.Cleanup(func() { noError(t, ts.alice.StopJob(ctx, jobID), "Cleanup stop job.") })
+
+		w := &strings.Builder{}
+		noError(t, ts.alice.StreamLogs(ctx, jobID, w), "Stream logs.")
+		// We expect to be pid 1.
+		assert(t, "1", strings.TrimSpace(w.String()), "invalid pid")
+	})
+}
+
+func TestNetworkNamespace(t *testing.T) {
+	t.Parallel()
+
+	ts, ctx := newTestServer(t)
+
+	t.Run("iface", func(t *testing.T) {
+		t.Parallel()
+
+		jobID, err := ts.alice.StartJob(ctx, "ip", []string{"address", "show"})
```

Comment:
Check the status before other assertions? The `ip` command isn't present on all systems.

Reply:
Done.